pylearn2/linear/tests/test_cudnn.py

```diff
@@ -0,0 +1,151 @@
+"""
+Tests for the Cudnn code
+"""
+
+import theano
+from theano import tensor
+from theano.sandbox.cuda.dnn import dnn_available
+
+from pylearn2.linear.conv2d import Conv2D
+from pylearn2.linear.cudnn2d import Cudnn2D, make_random_conv2D
+from pylearn2.space import Conv2DSpace
+from pylearn2.utils import sharedX
+from pylearn2.testing.skip import skip_if_no_gpu
+
+import unittest
+from nose.plugins.skip import SkipTest
+
+import numpy as np
+
+
+class TestCudnn(unittest.TestCase):
+    """
+    Tests for the Cudnn code
+    """
+    def setUp(self):
+        """
+        Set up a test image and filter to re-use
+        """
+        self.orig_floatX = theano.config.floatX
+        theano.config.floatX = 'float32'
```

Comment:
This needs to be done after the "SkipTest" exceptions can be raised, otherwise `tearDown` is not called, and the value is not restored.